src/components/connectAdvanced.js

```diff
@@ -100,203 +90,169 @@ export default function connectAdvanced(
 
     const displayName = getDisplayName(wrappedComponentName)
 
+    let PureWrapper
+
+    if (forwardRef) {
```

Comment:
I'm really not keen on this "if `forwardRef` generate A, else generate B" branches.  I know why they're here, but I think these can be simplified.